indexing-service/src/main/java/org/apache/druid/guice/IndexingServiceModuleHelper.java

```diff
@@ -37,5 +40,22 @@ public static void configureTaskRunnerConfigs(Binder binder)
     JsonConfigProvider.bind(binder, INDEXER_RUNNER_PROPERTY_PREFIX, RemoteTaskRunnerConfig.class);
     JsonConfigProvider.bind(binder, INDEXER_RUNNER_PROPERTY_PREFIX, HttpRemoteTaskRunnerConfig.class);
     JsonConfigProvider.bind(binder, "druid.zk.paths.indexer", IndexerZkConfig.class);
+
+    configureTaskIdentitiesProvider(binder);
```

Comment:
Is this binding needed here? It looks like this function is called in the middle manager, indexer and overlord, but it seems like it's only needed in the peon. Am I understanding that correctly?

Reply:
MM and indexer also run tasks thus this binding is needed.